src/function/scalar/operators/add.cpp

```diff
@@ -62,6 +69,9 @@ timestamp_t AddOperator::Operation(date_t left, dtime_t right) {
 	if (!Timestamp::TryFromDatetime(left, right, result)) {
 		throw OutOfRangeException("Timestamp out of range");
 	}
+	if (!Value::IsFinite(result)) {
```

Comment:
In `date_t AddOperator::Operation(date_t left, int32_t right)` we check for infinities before trying to add stuff, why is this not required here?

Reply:
Good catch! Adding a `TIME` to an infinite `DATE` should produce the same infinity.